### PR TITLE
reimbursable Journal Entry

### PR DESCRIPTION
**Pull request by anishfyle**

### CODE REVIEW

File: apps/sage_intacct/utils.py [hunk 1]
```diff
@@ -1036,7 +1036,7 @@ def __construct_journal_entry(self, journal_entry: JournalEntry, journal_entry_l
         for lineitem in journal_entry_lineitems:
             expense_link = self.get_expense_link(lineitem)
             credit_line = {
-                'accountno': general_mappings.default_credit_card_id,
+                'accountno': general_mappings.default_credit_card_id if general_mappings.default_credit_card_id else general_mappings.default_gl_account_id,
```
Comment (Shwetabhk) [inline apps/sage_intacct/utils.py:1039]: this is not right, if I am using both credit card and reimbursable as journal, we'll always be selecting the default_credit_card_id

we should computing this based on fund_source

Comment (anishfyle) [reply apps/sage_intacct/utils.py:1039]: yes @Shwetabhk, will fix this.

File: apps/sage_intacct/utils.py [hunk 2]
```diff
@@ -1063,6 +1063,35 @@ def __construct_journal_entry(self, journal_entry: JournalEntry, journal_entry_l
                 }
             }
 
+        if configuration.reimbursable_expenses_object == 'JOURNAL_ENTRY':
+            credit_line_re = {
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:1067]: separate dict is not needed, can just update the existing dict's accountno key.

File: apps/sage_intacct/utils.py [hunk 3]
```diff
@@ -1063,6 +1063,35 @@ def __construct_journal_entry(self, journal_entry: JournalEntry, journal_entry_l
...
+                'accountno': general_mappings.default_gl_account_id,
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:1068]: `'accountno': general_mappings.default_credit_card_id if journal_entry.expense_group.fund_source == 'CCC' else general_mappings.default_gl_account_id`

File: apps/sage_intacct/utils.py [hunk 4]
```diff
@@ -1110,9 +1139,11 @@ def __construct_journal_entry(self, journal_entry: JournalEntry, journal_entry_l
             for dimension in lineitem.user_defined_dimensions:
                 for name, value in dimension.items():
                     credit_line[name] = value
+                    credit_line_re[name] = value
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:1142]: not needed

File: apps/sage_intacct/utils.py [hunk 5]
```diff
@@ -1110,9 +1139,11 @@ def __construct_journal_entry(self, journal_entry: JournalEntry, journal_entry_l
...
                     debit_line[name] = value
 
             journal_entry_payload.append(credit_line)
+            journal_entry_payload.append(credit_line_re)
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:1146]: same

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**